file.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"fmt"
+	"net/http"
+	"io"
+	"path/filepath"
+	"bytes"
+	"archive/zip"
+	"strings"
+)
+
+// Download the zip file at the given URL to a temporary local directory.
+// Returns the asbolute path to the downloaded zip file.
+// IMPORTANT: You must call defer os.RemoveAll(dir) when done with this directory!
+func downloadGithubZipFile(repoOwner, repoName, gitTag, githubToken string) (string, *fetchError) {
+	// Create a temp directory
+	tempDir, err := ioutil.TempDir("", "")
+	if err != nil {
+		return "", newErr(err)
+	}
+
+	// Create an empty file to write to
+	file, err := os.Create(filepath.Join(tempDir, "repo.zip"))
```

Comment:
Why do you have to create an empty file? Can you note write the contents of `resp.Body` further down to a file directly with something like `ioutil.WriteFile`?